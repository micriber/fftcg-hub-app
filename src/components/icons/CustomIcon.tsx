import React from 'react';
import {Image, ImageSourcePropType, StyleSheet} from 'react-native';

type Props = {
  size?: {width: number; height: number};
  inactiveColor?: 'gray';
  src: ImageSourcePropType;
};

const CustomIcon = ({inactiveColor, src, size}: Props) => {
  const stylesExtended = {
    ...(inactiveColor ? {tintColor: inactiveColor} : {}),
  };

  return <Image style={[styles.image, size, stylesExtended]} source={src} />;
};

const styles = StyleSheet.create({
  image: {
    width: 20,
    height: 20,
  },
});

export default CustomIcon;
