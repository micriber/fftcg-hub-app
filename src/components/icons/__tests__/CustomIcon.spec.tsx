/**
 * @format
 */

import 'react-native';
import React from 'react';
import CustomIcon from '../CustomIcon';
import {render} from '@testing-library/react-native';

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Component] FFCardsList', () => {
  it('renders correctly', async () => {
    const root = render(
      <CustomIcon
        circle={false}
        src={require('../../../assets/icons/fire.png')}
      />,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders correctly in circle', async () => {
    const root = render(
      <CustomIcon
        circle={true}
        src={require('../../../assets/icons/fire.png')}
      />,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom size', async () => {
    const root = render(
      <CustomIcon
        circle={false}
        size={{width: 40, height: 40}}
        src={require('../../../assets/icons/fire.png')}
      />,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders correctly in circle with custom size', async () => {
    const root = render(
      <CustomIcon
        circle={true}
        size={{width: 40, height: 40}}
        src={require('../../../assets/icons/fire.png')}
      />,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });
});
