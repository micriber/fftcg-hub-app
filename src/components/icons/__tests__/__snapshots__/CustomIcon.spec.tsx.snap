// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] FFCardsList renders correctly 1`] = `
<Image
  source="test-file-stub"
  style={
    Array [
      Object {
        "height": 20,
        "width": 20,
      },
      Object {},
    ]
  }
/>
`;

exports[`[Component] FFCardsList renders correctly in circle 1`] = `
<Image
  source="test-file-stub"
  style={
    Array [
      Object {
        "height": 20,
        "width": 20,
      },
      Object {
        "borderColor": "#333",
        "borderRadius": 10,
        "borderWidth": 100,
      },
    ]
  }
/>
`;

exports[`[Component] FFCardsList renders correctly in circle with custom size 1`] = `
<Image
  source="test-file-stub"
  style={
    Array [
      Object {
        "height": 40,
        "width": 40,
      },
      Object {
        "borderColor": "#333",
        "borderRadius": 20,
        "borderWidth": 100,
      },
    ]
  }
/>
`;

exports[`[Component] FFCardsList renders correctly with custom size 1`] = `
<Image
  source="test-file-stub"
  style={
    Array [
      Object {
        "height": 40,
        "width": 40,
      },
      Object {},
    ]
  }
/>
`;
